### PR TITLE
Enable the eslint curly check

### PR DESCRIPTION
**Pull request by lindner**
- Run eslint fix on affected files
- Run clang-format on affected lines
- Update to latest clang-format

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -32,9 +32,6 @@ module.exports = {
     "max-len": "off",
     "comma-dangle": "off",
     "linebreak-style": "off",
-
-    // Things we might care about if we could automate them.
-    "curly": "off", // curly braces even single line blocks, --fix does something weird.
```
Comment (shaper) [inline .eslintrc.js]: Hoping whatever this was referencing in the past is now fixed, if @shans has opinion please comment.

Comment (lindner) [reply .eslintrc.js]: The problem was the eslint 'fixed' this by putting braces in weird places.  If you use clang-format you can automate the cleanup.

For example
```
  if (foo)
     return true;
```

would be rewritten as 
```
   if (foo)
     {return true;}
```

clang-format fixes that automatically

**Comment by lindner** [comment]
Any last minute objections.  Better get this in before there are more merge conflicts..

**Comment by shaper** [comment]
The vote supported, let's try it.

**Pull request merged**